core/shared/src/main/scala/scalaz/zio/Errors.scala

```diff
@@ -6,5 +6,6 @@ object Errors {
   final case class UnhandledError(error: Any, defects: List[Throwable] = Nil)
       extends Exception("An error was not handled by a fiber: " + error.toString())
 
-  final object TerminatedFiber extends Exception("The fiber was terminated either by a defect or an interruption")
+  final case class TerminatedFiber(causes: List[Throwable])
```

Comment:
I think we should try to be as precise as we can here:

```scala
final case class TerminatedFiber(defect: Throwable, defects: List[Throwable]) extends ...
final case class InterruptedFiber(causes: List[Throwable], defects: List[Throwable]) extends ...
```

Reply:
`unsafeRun` used to throw `defect` directly. Do you mean we should throw a `TerminatedFiber` encapsulating `defect` instead?